src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -281,6 +284,71 @@ public Predicate<Person> getCurrentPredicate() {
         return result;
     }
 
+    @Override
+    public ObservableList<Tag> getTagFiltersList() {
+        return tagFilters;
+    }
+
+    @Override
+    public ObservableList<RsvpStatus> getStatusFiltersList() {
+        return statusFilters;
+    }
+
+    @Override
+    public void addTagFilters(Set<Tag> tagFilters) {
+        for (Tag tag : tagFilters) {
+            if (!this.tagFilters.contains(tag)) {
+                this.tagFilters.add(tag);
+            }
+        }
+    }
+
+    @Override
+    public void addStatusFilters(Set<RsvpStatus> statusFilters) {
+        for (RsvpStatus status : statusFilters) {
+            if (!this.statusFilters.contains(status)) {
+                this.statusFilters.add(status);
+            }
+        }
+    }
+
+    @Override
+    public boolean checkTagFilterAlreadyExists(Tag tagToCheck) {
+        if (this.tagFilters.contains(tagToCheck)) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean checkStatusFilterAlreadyExists(RsvpStatus statusToCheck) {
+        if (this.statusFilters.contains(statusToCheck)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public void removeFilters(Set<Tag> tagFilters, Set<RsvpStatus> statusFilters) {
+        this.tagFilters.remove(tagFilters);
+        this.statusFilters.remove(statusFilters);
+
+        for (Tag tag : tagFilters) {
+            this.tagFilters.remove(tag);
+        }
+
+        for (RsvpStatus status : statusFilters) {
+            this.statusFilters.remove(status);
+        }
```

Comment:
Might be a bit confusing what each of the for loop does since the filters are already removed in the `this.tagFilters.remove(tagFilters);` and `this.statusFilters.remove(statusFilters);`